### ISSUE TITLE
Mypy shows no errors after previous run with errors.

### ISSUE THREAD

**Issue by mrolle45** [python/mypy#12428]
**Bug Report**

I run `mypy --no-incremental` and get lots of errors reported.
Then I run plain `mypy` and it says "no issues found"

**To Reproduce**

(Write your steps here:)

1.
```
D:\Solutions\trunk\Python\Sass>mypy --no-incremental .
solver\debugging.py:55: error: Name "InstrCtx" is not define
(90 more errors)...
Found 91 errors in 20 files (checked 38 source files)
```
2.
```
D:\Solutions\trunk\Python\Sass>mypy .
Success: no issues found in 38 source files
```
3.
```
D:\Solutions\trunk\Python\Sass>mypy .
Success: no issues found in 38 source files
```
**Expected Behavior**
Same output all 3 times.
**Actual Behavior**
See above.
**Comments**
When running again with `pytype -v`, the log shows:

- Found source:  (for each of 38 source files)
- Metadata fresh for:  (for each of 38 source files, then lots of other files)
- Loaded graph with 139 nodes (0.239 sec)
- Info about 73 SCCs.
- 73 fresh SCCs (139 nodes) left in queue (and will remain unprocessed)
- Build finished in 0.264 seconds with 0 modules, and 0 errors

_There is no type checking of anything!_

To me, the metadata files are indeed all fresh.
However, the Builder has no way of knowing that 20 of the source files contain errors.
I would suggest:

- Storing the status of errors found somewhere in the cache.
- The {module}.meta.json file would be a good place.
- Indicate how many errors were found, and (if any), whether there were any blockers.
- If there are not an unreasonable number of errors, store all the error messages.
- The Builder will check the error count.  If messages are stored in the metadata, then report them.
- Otherwise, run a type check on the module and report the errors.
- No need to write new metadata.
- However, if the metadata has a different mypy version ID, the metadata should be considered stale.  (Maybe this already happens anyway).

If I find the time to implement this suggestion myself and it works correctly, I'll add a comment with the fix.

**Your Environment**

- Mypy version used: 0.931
- Mypy command-line flags: --no-interactive (the first time)
- Mypy configuration options from `mypy.ini` (and other config files):
```
# Configuration for running mypy on this project.

[mypy]

exclude = (?x)(
    # Any top level name EXCEPT (given name)

    ^(?!(cache|metaclasses|outfile|asm|disasm|solver|sm_52)[./])
  )
ignore_errors = False
show_traceback = True
no_color_output = True

[mypy-solver.*]
ignore_errors = False
```

- Python version used: 3.7
- Operating system and version: Windows 10

**Comment by ilevkivskyi** [comment]
IIUC this is essentially the same as https://github.com/python/mypy/issues/12637 which is now fixed. Also we now store errors as part of the cache metadata, so we can use the cached data even if there were errors.
